### PR TITLE
Prac 05 feedback

### PR DESCRIPTION
**Pull request by talitha1234**
Could you please review my code, in particular, where I've sorted my file lines in functions in the wimbledon program? @jessepurcell

### CODE REVIEW

File: prac_05/wimbledon.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    """
+    with open(filename, "r", encoding="utf-8-sig") as in_file:
+        # read first line in file to skip and get rest of lines
+        in_file.readline()
+        lines = [line.strip().split(',') for line in in_file]
+
+        champion_to_win_count = calculate_champion_to_win_count(lines)
+        display_champion_to_win_count(champion_to_win_count)
+
+        display_champion_countries(lines)
```
Comment (jessepurcell) [inline prac_05/wimbledon.py]: Should the file remain open after reading the file?

File: prac_05/wimbledon.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        champion_to_win_count[champion] = champion_to_win_count.get(champion, 0) + 1
+    return champion_to_win_count
+
+
+def display_champion_countries(lines):
+    """Displays champion country codes"""
+    # the countries of the champions in alphabetical order
+    # set to remove duplicates, sorted in ABC order join it all with spaces between
+    countries = set(f'{line[1]}' for line in lines)
+    champion_to_country_code = " ".join(sorted(countries))
```
Comment (jessepurcell) [inline prac_05/wimbledon.py]: Maybe the comment should be above the line it is referencing?

File: prac_05/wimbledon.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+def calculate_champion_to_win_count(lines):
+    """Champions and how many times they have won. (List of lists and dictionary)"""
+    champions = [line[2] for line in lines]
+    # count number of times they were champions
+    #  dictionary
+    champion_to_win_count = {}
+    for champion in champions:
+        # add the champion
+        # get returns number of wins for champion or default to zero if no wins counted yet
```
Comment (jessepurcell) [inline prac_05/wimbledon.py]: Possibly redundant comments as the code is self documenting

File: prac_05/wimbledon.py (hunk 2 above)
Comment (jessepurcell) [inline prac_05/wimbledon.py]: Name of variable `champion_to_country_code `implies that this is a dictionary when it's a string
Maybe inline the `string.join` to the print statement on line 51 as it isn't used for anything else?

File: prac_05/wimbledon.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
+"""
+Word Occurrences
+Estimate: 120 mins
+Actual: roughly 2 or more hours split up over a few days
+"""
+filename = 'wimbledon.csv'
```
Comment (jessepurcell) [inline prac_05/wimbledon.py]: Is this a global variable or a constant? 🤔

File: prac_05/wimbledon.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
+"""
+Word Occurrences
```
Comment (jessepurcell) [inline prac_05/wimbledon.py:2]: Copy and paste?

File: prac_05/emails.py
Comment (jessepurcell) [inline prac_05/emails.py:1]: Looks good. Like the way you chose to grab the name from the email.

File: prac_05/hex_colours.py
Comment (jessepurcell) [inline prac_05/hex_colours.py:1]: I love everything about this file!

File: prac_05/wimbledon.py (hunk 1 above)
Comment (jessepurcell) [reply prac_05/wimbledon.py]: More clarification, should you be doing all this extra work inside the `with` block?

File: prac_05/wimbledon.py (hunk 5 above)
Comment (talitha1234) [reply prac_05/wimbledon.py:2]: I looked at one of the 3 examples from the lecture slides with the same problem and rewrote that for my first loop. Please see my updated code where I've demonstrated the LBYL method (originally EAFP) for my first loop and a list comprehension for my second loop.

File: prac_05/wimbledon.py (hunk 1 above)
Comment (talitha1234) [reply prac_05/wimbledon.py]: I see what you're saying. My functions don't need to be in line with the with statement so I've fixed this.

File: prac_05/wimbledon.py (hunk 2 above)
Comment (talitha1234) [reply prac_05/wimbledon.py]: Makes sense. I've moved the comment about the 2nd line of code so that this is more clear.

Comment (talitha1234) [reply prac_05/wimbledon.py]: I can see how that would be the incorrect naming convention.
That makes sense putting it directly in the print statement. I don't need the variable.

**Pull request merged**